### ISSUE TITLE
Some response security headers are not coming through when applied to traefik dashboard

### ISSUE THREAD

**Issue by andrewsav-datacom** [containous/traefik#5568]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [ X] No

This issue was discussed with and demonstrated to Daniel Tomcej @dtomcej before posting here.

### What did you do?

docker-compose.yaml:

```yaml
version: "3.3"
services:
  traefik:
    image: "traefik:v2.0.0"
    container_name: "traefik"
    command:
      - --entryPoints.web.address=:80
      - --log.level=DEBUG
      - --accesslog
      - --api
      - --providers.docker.exposedByDefault=false
      - --providers.file.filename=/dyn.yaml
    ports:
      - "80:80"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "./dyn.yaml:/dyn.yaml"
    labels:   
      - traefik.enable=true
      - traefik.http.routers.traefik-dash2-unsecure.rule=PathPrefix("/")
      - traefik.http.routers.traefik-dash2-unsecure.entrypoints=web
      - traefik.http.routers.traefik-dash2-unsecure.middlewares=testfirefoxheaders@file
      - traefik.http.routers.traefik-dash2-unsecure.service=api@internal
  whoami:
    image: containous/whoami
    labels:
      #- "traefik.enable=true"     
      - "traefik.http.routers.whoami_http.rule=PathPrefix(`/`)"
      - "traefik.http.routers.whoami_http.entrypoints=web"
      - traefik.http.routers.whoami_http.middlewares=testfirefoxheaders@file
```

dyn.yaml:

```yaml
http:
  middlewares:
    testfirefoxheaders:
      headers:
        browserXssFilter: true
        contentTypeNosniff: true
        frameDeny: true
```

log excerpt

```log
traefik    | time="2019-10-02T21:15:18Z" level=debug msg="Creating middleware" middlewareName=testfirefoxheaders@file middlewareType=Headers entryPointName=web routerName=traefik-dash2-unsecure@docker
traefik    | time="2019-10-02T21:15:18Z" level=debug msg="Setting up secureHeaders from %v{map[] map[] false [] []  [] 0 false [] [] false false  map[] false 0 false false false true  true true      false}" entryPointName=web routerName=traefik-dash2-unsecure@docker middlewareName=testfirefoxheaders@file middlewareType=Headers
```

Abridged output of `curl -i -vvv http://127.0.0.1/dashboard/`

```text
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 80 (#0)
> GET /dashboard/ HTTP/1.1
> Host: 127.0.0.1
> User-Agent: curl/7.47.0
> Accept: */*
>
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Accept-Ranges: bytes
Accept-Ranges: bytes
< Content-Length: 3403
Content-Length: 3403
< Content-Type: text/html; charset=utf-8
Content-Type: text/html; charset=utf-8
< Last-Modified: Mon, 16 Sep 2019 17:18:10 GMT
Last-Modified: Mon, 16 Sep 2019 17:18:10 GMT
< Date: Wed, 02 Oct 2019 21:16:14 GMT
Date: Wed, 02 Oct 2019 21:16:14 GMT

<
<!DOCTYPE html>
```

### What did you expect to see?

I expected to see the response headers coming through:

``` text
< X-Content-Type-Options: nosniff
< X-Frame-Options: DENY
< X-Xss-Protection: 1; mode=block
```

### What did you see instead?

See above, them NOT coming through. 

Note that this is only issue with the dashboard. The headers do come through for other services as can be demonstrated by commenting out the first `traefik.enable=true` and uncommenting the second.

Some more discussion here: https://community.containo.us/t/middlewares-questions-and-suggestions/1957/6

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Traefik version 2.0.0 built on 2019-09-16T17:35:17Z
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

See above.

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

See above

**Comment by dtomcej** [comment]
Same is reproducible on `2.0.1`

**Comment by 0xbb** [comment]
Same for `2.0.5`

**Comment by matej-hartmann** [comment]
I can confirm this on 2.0.5, also the tls.options are ignored. Basic auth works fine though, so maybe this is only affecting SSL/TLS related options?

**Comment by iamcryptoki** [comment]
Same issues for 2.1.

**Comment by Clownfused** [comment]
Having this same issue on v2.2.0-rc2

**Comment by smares** [comment]
Just noticed this issue with 2.2.1 using the following config:

```
http:
  middlewares:
    hsts-headers:
      headers:
        stsSeconds: 63072000
        stsIncludeSubdomains: true
        stsPreload: true
    security-headers:
      headers:
        frameDeny: true
        contentTypeNosniff: true
        browserXssFilter: true
```

My docker-compose.yml contains the following label:

"traefik.http.routers.traefik.middlewares=authenticate,compress,hsts-headers@file,security-headers@file"

The authenticate and compress middlewares are also defined per label.

According to the dashboard, all 4 middlewares are active for the router.

**Comment by virtualben89** [comment]
> Just noticed this issue with 2.2.1 using the following config:
> 
> ```
> http:
>   middlewares:
>     hsts-headers:
>       headers:
>         stsSeconds: 63072000
>         stsIncludeSubdomains: true
>         stsPreload: true
>     security-headers:
>       headers:
>         frameDeny: true
>         contentTypeNosniff: true
>         browserXssFilter: true
> ```
> 
> My docker-compose.yml contains the following label:
> 
> `"traefik.http.routers.traefik.middlewares=authenticate,compress,hsts-headers@file,security-headers@file"`
> 
> The authenticate and compress middlewares are also defined per label.
> 
> According to the dashboard, all 4 middlewares are active for the router, but neither the HSTS headers are sent, nor the security headers.
> 
> Edit: In the meanwhile, I realized that security headers in multiple middlewares cannot be merged, but even with all headers set inside a single middleware, they are not applied to the dashboard router.

I can confirm this issue. Setting the force STS ends in SSL Labs Test Grade B, without force STS it´s an A.

**Comment by traefiker** [comment]
Closed by #6750.
